src/internal/util/pipe.ts

```diff
@@ -25,6 +26,10 @@ export function pipeFromArray<T, R>(fns: Array<UnaryFunction<T, R>>): UnaryFunct
     return noop as UnaryFunction<any, any>;
   }
 
+  if (fns.length === 0) {
+    return identity as UnaryFunction<any, any>;
+  }
+
```

Comment:
IMO, the previous test should be included in this `if` statement and both should return `identity`:

```
  if (!fns || fns.length === 0) {
    return identity as UnaryFunction<any, any>;
  }
```
I can see no reason whatsoever for returning a no-op. Having different behaviour for no-array and empty-array scenarios makes little sense.

I think you should do what it was that you mentioned in the issue you raised and return `identity`. The code might not be reachable ATM, but if the `pipeFromArray` is used elsewhere in the future, it should behave in a predictable way.

Reply:
fixed it myself.